### PR TITLE
Add DataPointList for Scatterplot, to add a label for each marker.

### PR DESCRIPTION
**Pull request by Cubikon**
#2141

### CODE REVIEW

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (swharden) [inline src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:430]: Let's use { } to wrap the full contents of this if statement

Comment (swharden) [reply src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:430]: Also, this if statement should be _outside_ the if statement two lines up about markers. I think it makes sense to always show the labels if they are populated (not only if marker size is >0)

**Review by swharden — CHANGES_REQUESTED** [review]

**Comment by swharden** [comment]
This looks great @Cubikon, thanks! If you make the change I requested I'll merge this in good time 👍 🚀

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (bclehmann) [inline src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:426]: I think this should be `DataPointLabels is not null`.

In this case they're the same, but theoretically if we swapped out the type of `DataPointLabels` it is possible we could get different results. And while overloading `==` in such a way as to change the correctness is a little absurd, we could still get (marginally) better performance. If `==` is overloaded the compiler will have to call the overload when we use `==`, rather than just emitting a `ceq` opcode when we use `is`.

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (bclehmann) [inline src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:433]: Perhaps this should be customizable? Although I'm happy leaving this simple and adding customization if and when it's necessary.

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (bclehmann) [inline src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:434]: This variable is unused

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (bclehmann) [inline src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:436]: I believe `textFont.Rotation` is always zero, so this line is redundant (unless we decide to add customization later)

**Review by bclehmann — CHANGES_REQUESTED** [review]
Looks pretty good, mostly just some minor cleanup around unnecessary code.

**Comment by bclehmann** [comment]
@swharden If you think my change requests are too minor to block merging over than feel free to push it through and I can make a small PR later with them.

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (swharden) [reply src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:433]: We might as well expose it as a public field `DataPointLabelFont`

I'd make the change myself and merge it now but I don't have write access to the branch 😅

File: src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs
Comment (swharden) [reply src/ScottPlot4/ScottPlot/Plottable/ScatterPlot.cs:436]: If we expose `DataPointLabelFont` as described earlier, rotation could be customized, so let's leave this like it is 👍

**Comment by swharden** [comment]
> @swharden If you think my change requests are too minor to block merging over than feel free to push it through and I can make a small PR later with them.

I'd make the changes myself and merge but I don't have write access to the branch...

If we don't hear back from @Cubikon by tomorrow night I'll merge it as it is 👍

**Comment by Cubikon** [comment]
i did the new requested changes. And you have now access to the branch, if you want to change something else.

**Review by swharden — APPROVED** [review]

**Comment by swharden** [comment]
Looks great, thanks! 🚀

**Pull request merged**
